### PR TITLE
Shahein handlebars cleanup

### PR DESCRIPTION
**Pull request by shaheinm**
So it looks good now when I run it - the carousel is working on the home page, the nav bar is on every page, and the links on the nav bar work.

The solution to get back to the profile page is a bit hacky -- absent true authentication (which, I don't care that much about right now), I stored the user ID in localStorage after registration.  I used that key in localStorage to complete the `/profile/:id` route so express-handlebars could render the appropriate profile with the right data.

All the pictures are working now.  Any time you reference something in the `public` folder (images, css, javascript, etc), you just need to use it's path as if you were already in the `public` folder.  This is because of the line in your server - `app.use(express.static('public'));` - which serves the public folder as if it were at the root of your application.

Pull this code down, run it, and let me know if you have any questions.

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -36,9 +36,12 @@ router.get('/profile/:id', (req, res) => {
...
         }
     }).then((data) => {
-        // console.log("this is data", data);
-        console.log("this is data", data.zipcode);
         res.render('profile', {
+            fullname: data.firstName + " " + data.lastName,
+            firstname: data.firstName,
+            lastname: data.lastName,
+            phone: data.mobile,
+            address: data.address,
```
Comment (shaheinm) [inline controllers/userController.js]: Passing all the profile data you have right now.  You'll likely need to add email, education and work experience to it (in the same table or in separate ones) if you want to put that data on the profile page.  Also, it looks great!

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "main": "config.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node server.js"
+    "start": "node server.js",
+    "dev": "nodemon server.js"
```
Comment (shaheinm) [inline package.json]: Added this and nodemon to your devDependencies for an easier time working with it.  You don't need to globally install nodemon if you haven't already - just run `npm run dev`

File: public/assets/js/client.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ function postSignUpData(person) {
       .then((data) => {
           console.log("Success --> ", data);
           const userId = data.message;
+          localStorage.setItem('userId', userId);
```
Comment (shaheinm) [inline public/assets/js/client.js]: HACK

File: views/home.handlebars [hunk 4]
```diff
@@ -1,58 +1,3 @@
...
-            <li><a href="#map">MAP</a></li>
-
-            <li>
-              <a href="#"><span class="glyphicon glyphicon-search"></span></a>
-            </li>
-          </ul>
-        </div>
-      </div>
-    </nav>
-
```
Comment (shaheinm) [inline views/home.handlebars]: Don't need all this html stuff, that's in the `main.handlebars` layout.

File: views/layouts/main.handlebars [hunk 5]
```diff
@@ -2,9 +2,79 @@
...
+
+            <li>
+              <a href="#"><span class="glyphicon glyphicon-search"></span></a>
+            </li>
+          </ul>
+        </div>
+      </div>
+    </nav>
+
+    {{{ body }}}
```
Comment (shaheinm) [inline views/layouts/main.handlebars]: All of your templates are rendered as if you wrote the html write here.  The rest of the stuff on this page is on *every* page now, creating a consistent layout.  And again, it looks nice!

File: views/profile.handlebars [hunk 6]
```diff
@@ -1,31 +1,6 @@
...
-        <nav class="navbar navbar-default">
-          <div class="container-fluid">
-            <div id="navbar" class="navbar-collapse collapse">
-              <ul class="nav navbar-nav">
-                <li class="active"><a href="./profile.handlebars">Home</a></li>
-              </ul>
-            </div><!--/.nav-collapse -->
-          </div><!--/.container-fluid -->
+<div class="container">
+<a href src="/socket/index.html">Socket Chat App</a>
```
Comment (shaheinm) [inline views/profile.handlebars]: Don't know what this is really, but thought it'd be easier to reference if I moved it to the `public` folder.  It's static, whatever it is, so it should work.

File: views/profile.handlebars [hunk 7]
```diff
@@ -34,17 +9,17 @@ html,body,h1,h2,h3,h4,h5,h6 {font-family: "Roboto", sans-serif}
 
           <div class="w3-white w3-text-grey w3-card-4">
             <div class="w3-display-container">
-              <img src="../public/assets/images/butterfly.jpg" style="width:100%" alt="profile">
-              <div class="w3-display-bottomleft w3-container w3-text-black">
-                <h2>Jane Doe</h2>
+              <img src="/assets/images/butterfly.jpg" style="width:100%" alt="profile">
+              <div class="w3-display-bottomleft w3-container w3-text-white">
```
Comment (shaheinm) [inline views/profile.handlebars]: Changed to white so you can see it on the butterfly photo.

File: views/signup.handlebars [hunk 8]
```diff
@@ -19,11 +8,8 @@
                             href="/">Sign In</a></div>
                 </div>
                 <div class="panel-body">
-                    <form class="form-horizontal" role="form" method="post" action="/signup"
-                        onSubmit="return checkblank(this);">
-                        <% if (message.length > 0) { %>
-                        <div class="alert alert-success col-sm-12"><%= message %></div>
-                        <% } %>
+                    <form class="form-horizontal" role="form" method="post" action="/signup" onSubmit="return checkblank(this);">
+                        <div class="alert alert-success col-sm-12">Please fill out the form below:</div>
```
Comment (shaheinm) [inline views/signup.handlebars]: I just replaced the non-handlebars templating stuff with a standard message.  Feel free to modify.

File: controllers/userController.js (hunk 1 above)
Comment (KingdomB) [reply controllers/userController.js]: Approved.

**Pull request merged**